File: www/cgi/php-cgi/simple_cgi_script.php
#!/usr/bin/php
<?php
// Read the value of 'name' parameter from the query string
parse_str($_SERVER['QUERY_STRING'], $query);
$name = isset($query['name']) ? urldecode($query['name']) : '';

// Output HTML response with the formatted name
echo "Content-type: text/html\r\n\r\n";
?>
<!DOCTYPE html>
<html>
<head>
<title>Hello, <?php echo htmlspecialchars($name); ?>!</title>
<style>
body {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  font-family: Arial, sans-serif;
}
div {
  text-align: center;
}
</style>
</head>
<body>
<div>
<h1>Hello, <?php echo htmlspecialchars($name); ?>!</h1>
</div>
</body>
</html>

File: www/cgi/python-cgi/simple_cgi_script.py
#!/usr/bin/python3

import cgi
import cgitb; cgitb.enable()  # This enables CGI error reporting
from html import escape  # Import the escape function from the html module
import os  # Import the os module to access environment variables

def main():
    form = cgi.FieldStorage()  # Get data from the form
    name = form.getvalue('name', 'World')  # Get 'name' parameter
    print(f"""Content-type: text/html\r\n
    <!DOCTYPE html>
    <html>
    <head>
        <title>Hello, {escape(name)}!</title>
        <style>
            body {{
                display: flex;
                justify-content: center;
                align-items: center;
                height: 100vh;
                font-family: Arial, sans-serif;
            }}
            div {{
                text-align: center;
            }}
        </style>
    </head>
    <body>
        <div>
            <h1>Hello, {escape(name)}!</h1>
        </div>
    </body>
    </html>
    """)  # Print HTML response

if __name__ == "__main__":
    main()